test/integration/integration/utils/kubernetes/helpers.go

```diff
@@ -95,6 +95,40 @@ func WaitForGatewayAddress(t *testing.T, c client.Client, timeoutConfig config.T
 	return net.JoinHostPort(ipAddr, port)
 }
 
+// WaitForAPIListenerAddress waits until at least one IP Address has been set in the
+// Gateway infra exposed service.
+func WaitForAPIListenerAddress(t *testing.T, c client.Client, timeoutConfig config.TimeoutConfig) string {
```

Comment:
Shall we move the common logic in both `WaitForGatewayAddress` and `WaitForAPIListenerAddress` to another function and call it with lister port from two of these functions?

Reply:
done, resolved